File: find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
         List<Integer> l1 = new ArrayList<Integer>();
         Set<Integer> hash_Set = new HashSet<Integer>();
        for(int i=0; i<nums.length; i++){
            hash_Set.add(nums[i]);
        }
        for(int i=1; i<=nums.length; i++){
            if(!hash_Set.contains(i)){
                l1.add(i);
            }
        }
        return l1;
    }
}

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        ans = []
        for index in range(len(nums)):
            if nums[abs(nums[index]) - 1] > 0:
                nums[abs(nums[index]) - 1] *= -1

        for index in range(len(nums)):
            if nums[index] > 0:
                ans.append(index + 1)

        return ans